src/lib/analysers/meta/main.py

```diff
@@ -0,0 +1,71 @@
+from lib.common.analyser import Analyser
+from lib.common.analyser import paths_to_components
+from lib.common.get_module import get_module
+import os
+from shutil import copyfile, rmtree
+
+
+class MetaAnalyser(Analyser):
+    def setup_run(self, config):
+
+        self.child_analysers = []
+        whitelist = self.CONFIG["elements_in"]
+
+        cmps = paths_to_components(whitelist)
+        self.selector = cmps[0][0]
+
+        child_modules = config["modules"]
+        for module in child_modules:
+            child_name = module["name"]
+            child_config = module["config"]
+            ChildAnalyser = get_module("analyser", child_name)
+            if ChildAnalyser is None:
+                raise Exception(
+                    f"The module you have specified, {child_name}, does not exist"
+                )
+            child_analyser = ChildAnalyser(child_config, child_name, self.FOLDER)
+            child_analyser.setup_run(child_config)
+            self.child_analysers.append(child_analyser)
+            self.logger(f"Setup child analyser {child_name}")
+
+    def analyse_element(self, element, config):
+        src = None
+        child_element = None
+        for index, analyser in enumerate(self.child_analysers):
+            child_element = self._derive_child_element(index, element, src, analyser)
+            analyser.analyse_element(child_element, analyser.CONFIG)
+            el_id = element["id"]
+            self.logger(f"Analysed element {el_id} in {analyser.NAME}")
+            src = child_element["dest"]
+        self._finalise_element(config, child_element, element)
+
+    def post_analyse(self, config):
+        delete_cache = config["delete_cache"]
+        if delete_cache:
+            self.logger("deleting cache")
```

Comment:
Slightly more descriptive error message here is warranted, I think.

Reply:
Now logs the path of the cache to be deleted. Was this what you had in mind or more fine-grained logging of files and folders being deleted?

Reply:
Generic error logging is okay at the moment, just good to indicate which path it's failed on, yep.